### PR TITLE
feat(ui): add opt-in animated status indicators

### PR DESCRIPTION
**Pull request by JJLiebig**
## Summary

- restore animated working-agent indicators as a third `ui.status_indicators` option
- keep `dots` as the default and animate only while the desktop sidebar is visible and a connected agent is working
- keep animation entirely in the client renderer; no server or wire-protocol changes

Discussion that requested the feature: https://github.com/herdrdev/herdr/discussions/2534 

## Validation

- `just lint`
- focused animation and config parsing tests
- 3,100 Rust tests passed with the unrelated existing Windows ACL test `backup_preserves_legacy` excluded; that test failed identically in three standalone reruns
- 106 maintenance tests, UI hot-path architecture checks, integration asset tests, and docs contract tests

## Performance

`just bench-render-scale` remained effectively flat across 1 and 15 populated panes (client shell composition: 1.08x background, 0.89x active in this run).

On a native Windows release build with one working agent, three 15-second samples showed median client CPU of 3.65% animated versus 0.83% dots and client terminal-output traffic of 8.9 KB/s versus 6.6 KB/s. Server output traffic was unchanged. The feature therefore remains explicitly opt-in.

### CODE REVIEW

**Review by ogulcancelik — DISMISSED** [review]
the ~2 percentage point increase in client cpu feels too expensive for a spinner. i don't think we should accept that cost yet.

could we record visible spinner cells during normal rendering and patch only those on animation ticks, instead of rebuilding the whole sidebar?

80 ms means 12.5 fps, which doesn't seem excessive. i'd like to see where the time goes: sidebar composition, patch encoding, or terminal writes. the sidebar is a small part of the screen, so this much overhead needs explaining.

File: src/client/shell/state.rs [hunk 1]
```diff
@@ -1908,11 +1912,52 @@ impl ClientShellState {
...
+            && self
+                .last_composed_size
+                .is_some_and(|(cols, rows)| self.layout(cols, rows).sidebar.width > 0)
+            && self.endpoints.iter().any(|endpoint| {
+                endpoint.status == ClientEndpointStatus::Online
+                    && endpoint.snapshot.as_deref().is_some_and(|snapshot| {
+                        snapshot.agents.iter().any(|agent| {
+                            agent.agent_status == crate::api::schema::AgentStatus::Working
+                        })
+                    })
```
[the remark this answers is not in the document]
Comment (JJLiebig) [reply src/client/shell/state.rs]: Fixed in 2157c50e. Spinner eligibility now derives from the rows rendered by the last composition, so collapsed or scrolled-out working agents no longer keep the 80 ms timer alive. The focused test covers a hidden working agent and the collapsed-sidebar raw-status edge case.
